### ISSUE TITLE
[BUG] Agent Role in connection with CME

### ISSUE THREAD

**Issue by DominikBauer1** [tribe29/ansible-collection-tribe29.checkmk#232]
Verify first that your issue is not already reported [here](https://github.com/tribe29/ansible-collection-tribe29.checkmk/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc).
Where possible also test if the latest release and main branch are affected too.
Complete all sections as described!

**Describe the bug**

Theres are multiple issues when using the agent role with CME. 

This might be related to #65. However, after reading the [README](https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/main/roles/agent/README.md?#L21), it looks like the agent role will work with CME, so I opened this issue.

1. When running the role against a Monitoring site, managed by OME, the role is unable to download an agent. 
2. Using the "checkmk_agent_add_host" feature against a remote site creates an inconsistency.

**Component Name**
agent role 

**Ansible Version**

```console
$ ansible --version

ansible [core 2.13.6]

```

**Checkmk Version**

```console
2.1.0p17 (CME)
```

**Collection Version**

```console
$ ansible-galaxy collection list

```
**Environment**

**To Reproduce**
1. Create an central  and an remote site on two different  systems
4. Create playbook to completely add and register a (linux) host 
```console
---
- hosts: <some host>
  vars:
    checkmk_agent_edition: cme
    checkmk_agent_protocol: https
    checkmk_agent_server: <remote site  server>
    checkmk_agent_server_validate_certs: "true"
    checkmk_agent_site: <remote site>
    checkmk_agent_user: cmkadmin
    checkmk_agent_pass: <some pass>
    checkmk_agent_add_host: "true"
    checkmk_agent_discover: "true"
    checkmk_agent_update: "true"
    checkmk_agent_tls: "true"
    checkmk_agent_folder: "/emea"
    checkmk_agent_version: "2.1.0p17"

  roles:
      - tribe29.checkmk.agent
```
6. Execute 
7. See error

**Expected behavior**
The agent  gets downloaded and installed 

**Actual behavior**
The role fails with : 

```console
  json:
    detail: No baked agent for this host available
    status: 500
    title: An exception occurred
```

![image](https://user-images.githubusercontent.com/46919439/213203912-ca4d754b-4ee8-4083-beee-9006aad41dfb.png)

**Screenshots**
It seems like that the CME remote site does not provide any agent on the Monitoring site. Not even the GENERIC agent.

Remote site:
![image](https://user-images.githubusercontent.com/46919439/213194764-4303a527-eeb7-4640-a7a1-7f2aca3a660f.png)

**Additional context**

A possible solution would be to allow additional connection data when the CME edition is selected. Then it would be possible to run some tasks always  against the central site.  In my test it works as expected when Running  "[Create host on server.](https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/main/roles/agent/tasks/main.yml#L48)" and "[Debian Derivatives: Download GENERIC Checkmk Agent](https://github.com/tribe29/ansible-collection-tribe29.checkmk/blob/main/roles/agent/tasks/Debian.yml#L52)" against the central site.

**Comment by robin-tribe29** [comment]
HI @DominikBauer1 and thanks for reporting this.
#65 is not really related here, as that issue aims at the modules, and more specifically at the `customer` attribute, which needs to be handled.
#223 is the PR you might want to check out. There is still some fine-tuning to be done, but you can check out those changes and provide your feedback, if you like.

**Comment by robin-tribe29** [comment]
This should be fixed in [0.16.0](https://github.com/tribe29/ansible-collection-tribe29.checkmk/releases/tag/v0.16.0).

**Comment by robin-tribe29** [comment]
> I have a problem with that static localhost.
> https://github.com/tribe29/ansible-collection-tribe29.checkmk/commit/7774828493d4e2e5745a59c0ea8b85cac3f5c170#r99633338

Fixed by #262.
